### PR TITLE
Modify the Soak tests for Java

### PR DESCRIPTION
**Pull request by NathanielRN**
*Description of changes:*

Modifies the [AWS OTel Python Soaking Tests](https://github.com/aws-observability/aws-otel-python/blob/main/.github/workflows/soak-testing.yml) for using in the AWS OTel Java Instrumentation repo.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/soak-testing.yml [hunk 1]
```diff
@@ -32,21 +33,77 @@ env:
   # THROUGHPUT_PER_SECOND: TBD?
 
 jobs:
+  build_apps_to_ecr:
+    name: Publish Apps to ECR
```
Comment (anuraaga) [inline .github/workflows/soak-testing.yml]: Soak tests should not be publishing anything, only main (snapshot) and release build.

I think we need to parameterize this to specify `tags = ...:alpha` for the main-build and can reference alpha (latest snapshot) here

https://github.com/aws-observability/aws-otel-java-instrumentation/blob/main/sample-apps/spark-awssdkv1/build.gradle.kts#L24

Comment (anuraaga) [reply .github/workflows/soak-testing.yml]: Alternatively `jibDockerBuild` should work I think, but make sure to run it in the same `job` as the soak tests

Comment (NathanielRN) [reply .github/workflows/soak-testing.yml]: To speed up our solution here, we will "build" first and consider it a follow-up to "pull" the snapshot from the main build.

https://github.com/aws-observability/aws-otel-python/pull/26

**Comment by NathanielRN** [comment]
Tests ran overnight [on my fork](https://github.com/NathanielRN/aws-otel-java-instrumentation/actions/runs/1261534271) successfully.

Here are the snapshots from the 3 sample apps:

For spark:

CPU STATS
MEMORY STATS

Error distribution was concerning, but almost an order of magnitude less:

```
load-generator_1  | Status code distribution:
load-generator_1  |   [200]	999962 responses
load-generator_1  |   [404]	38 responses
load-generator_1  | 
load-generator_1  | Error distribution:
load-generator_1  |   [119668]	Get "http://app:8080/outgoing-http-call": dial tcp 172.18.0.2:8080: connect: connection refused
```

For spark-awssdkv1:

CPU STATS
MEMORY STATS

Error distribution was an order of magnitude less:

```
load-generator_1  | Status code distribution:
load-generator_1  |   [200]	1000000 responses
load-generator_1  | 
load-generator_1  | Error distribution:
load-generator_1  |   [99615]	Get "http://app:8080/outgoing-http-call": dial tcp 172.18.0.2:8080: connect: connection refused
```

For springboot:

CPU STATS
MEMORY STATS

One call out is that Springboot had _several_ dropped requests:

```
load-generator_1  | Status code distribution:
load-generator_1  |   [200]	999892 responses
load-generator_1  |   [500]	108 responses
load-generator_1  | 
load-generator_1  | Error distribution:
load-generator_1  |   [186285]	Get "http://app:4567/outgoing-http-call": dial tcp 172.18.0.2:4567: connect: connection refused
```

Based on these results, we selected a CPU Threshold of 40 and a higher memory threshold of 4 GiB.

**Review by anuraaga — APPROVED** [review]

**Comment by NathanielRN** [comment]
Closing this PR because we will merge these changes in https://github.com/aws-observability/aws-otel-java-instrumentation/pull/84. We leave this as a reference to see the changes that were needed to adapt the tests for Java.

**Pull request closed without merging**